### PR TITLE
TextPanel: Refactor to functional component

### PR DESCRIPTION
**Pull request by ryantxu**
This PR refactors the text panel from a class component into a functional component.

NOTE: this removes the debounce behavior because it is not required with `useMemo` and the monaco editor that only sends changes on onSave and onBlur

### CODE REVIEW

File: public/app/plugins/panel/text/module.tsx [hunk 1]
```diff
@@ -11,7 +11,6 @@ export const plugin = new PanelPlugin<PanelOptions>(TextPanel)
       .addRadio({
         path: 'mode',
         name: 'Mode',
-        description: 'text mode of the panel',
```
Comment (ryantxu) [inline public/app/plugins/panel/text/module.tsx:14]: removing the useless comments

File: public/app/plugins/panel/text/TextPanel.tsx [hunk 2]
```diff
@@ -1,127 +1,90 @@
...
-
-  updateHTML = debounce(() => {
-    const html = this.processContent(this.props.options);
-    if (html !== this.state.html) {
-      this.setState({ html });
-    }
-  }, 150);
+export function TextPanel({ options, replaceVariables, width, height }: Props) {
+  const styles = useStyles2(getStyles);
+  const location = useLocation();
```
Comment (ryantxu) [inline public/app/plugins/panel/text/TextPanel.tsx:16]: ^^ required so we get a callback when variables change

Comment (torkelo) [reply public/app/plugins/panel/text/TextPanel.tsx:16]: @ryantxu it should automatically get re-rendered if a variable that it includes in it's definition change. 

Is that not the case?

File: public/app/plugins/panel/text/TextPanel.tsx [hunk 3]
```diff
@@ -1,127 +1,86 @@
...
+        data-testid="TextPanel-converted-content"
+      />
+    </CustomScrollbar>
+  );
+}
 
-  prepareHTML(html: string): string {
-    const result = this.interpolateString(html);
-    return config.disableSanitizeHtml ? result : this.sanitizeString(result);
+export function processContent(
```
Comment (torkelo) [inline public/app/plugins/panel/text/TextPanel.tsx:52]: does not look like it's used outside this component (regard the export)

File: public/app/plugins/panel/text/TextPanel.tsx (hunk 2 above)
Comment (ryantxu) [reply public/app/plugins/panel/text/TextPanel.tsx:16]: Ahh -- the issue was that with `useMemo` there are no obvious props to invalidate.  I went back to a debouce approach, and have the invalidate whenever props change (not the options within it)

**Review by torkelo — APPROVED** [review]
Looks good!

**Pull request merged**